protecode/scanning.py

```diff
@@ -111,80 +111,101 @@ def iter_vulnerabilities_with_assessments(
-            logger.info(f'{group_name=}: did not find old scan')
-
-        if r.type is cm.ArtefactType.OCI_IMAGE:
-            def iter_content():
-                image_reference = r.access.imageReference
-                yield from oci.image_layers_as_tarfile_generator(
-                    image_reference=image_reference,
-                    oci_client=oci_client,
-                    include_config_blob=False,
-                    fallback_to_first_subimage_if_index=True,
+    ) -> typing.Generator[pm.ScanRequest, None, None]:
+        c = resource_group[0].component
+        r = resource_group[0].resource
+
+        group_id = _resource_group_id(resource_group)
+        group_name = f'{c.name}/{r.name}:{r.version} {r.type}'
+        known_results = known_artifact_scans.get(group_id)
+        display_name = f'{r.name}_{r.version}_{c.name}'.replace('/', '_')
+
+        if r.type is cm.ResourceType.OCI_IMAGE:
+            for resource_node in resource_group:
+                c = resource_node.component
+                r = resource_node.resource
+
+                # find product existing bdba scans (if any)
+                component_artifact_metadata = protecode.util.component_artifact_metadata(
+                    component=c,
+                    artefact=r,
+                    omit_component_version=False,
+                    omit_resource_version=False,
                 )
+                target_product_id = protecode.util._matching_analysis_result_id(
+                    component_artifact_metadata=component_artifact_metadata,
+                    analysis_results=known_results,
+                )
+                if target_product_id:
+                    logger.info(f'{group_name=}: found {target_product_id=}')
+                else:
+                    logger.info(f'{group_name=}: did not find old scan')
+
+                def iter_content():
+                    image_reference = r.access.imageReference
+                    yield from oci.image_layers_as_tarfile_generator(
+                        image_reference=image_reference,
+                        oci_client=oci_client,
+                        include_config_blob=False,
+                        fallback_to_first_subimage_if_index=True,
+                    )
 
-            return pm.ScanRequest(
-                component=c,
-                artefact=r,
-                scan_content=iter_content(),
-                display_name=display_name,
-                target_product_id=target_product_id,
-                custom_metadata=component_artifact_metadata,
-            )
+                yield pm.ScanRequest(
+                    component=c,
+                    artefact=r,
+                    scan_content=iter_content(),
+                    display_name=display_name,
+                    target_product_id=target_product_id,
+                    custom_metadata=component_artifact_metadata,
+                )
         elif (
             isinstance(r.type, str)
```

Comment:
nit (not actually part of this pull-request): why check for type - we might immediately check for equality w/ str-literal